Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs

```diff
@@ -22,34 +23,186 @@ public static void ShowHelp()
             Application.OpenURL("https://developer.microsoft.com/en-us/windows/holographic/unity_development_overview#Configuring_a_Unity_project_for_HoloLens");
         }
 
-        /// <summary>
-        /// Applies recommended scene settings to the current scenes
-        /// </summary>
-        [MenuItem("HoloToolkit/Configure/Apply HoloLens Scene Settings", priority = 0)]
-        public static void ApplySceneSettings()
+        [MenuItem("HoloToolkit/Configure/Apply All HoloLens Settings", priority = 0)]
+        public static void ApplyAllSettings()
         {
+            var output = new StringBuilder();
+            ApplySceneSettingsInternal(output);
+
+
+            var reload = ApplyProjectSettingsInternal(output);
+
+            EditorUtility.DisplayDialog("Results", output.ToString(), "Ok");
+
+            if (reload)
+            {
+                PromptForReload();
+            }
+        }
+
+
+        private static void ApplySceneSettingsInternal(StringBuilder output)
+        {
+            //todo if we find more than one camera, hololens will always use MainCamera (should)
+            //additional cameras could be used from a RenderTexture, etc. Should we warn?
+
+            output.AppendLine("** Checking Camera Settings");
+            Camera[] cameras = new Camera[Camera.allCamerasCount];
+            Camera.GetAllCameras(cameras);
+
             if (Camera.main == null)
             {
-                Debug.LogWarning(@"Could not apply settings - no camera tagged with ""MainCamera""");
+                output.AppendLine(@"Could not apply camera settings - no camera tagged with ""MainCamera""");
                 return;
             }
 
-            Camera.main.transform.position = Vector3.zero;
-            Camera.main.clearFlags = CameraClearFlags.SolidColor;
-            Camera.main.backgroundColor = Color.black;
-            Camera.main.nearClipPlane = 0.85f;
+            SetCameraDefaults(Camera.main, true, output);
+
+            //All done with MainCamera
+            cameras = cameras.Where(o => o != Camera.main).ToArray();
+
+            //Check remaining cameras
+            if (cameras.Length > 0)
+            {
+                foreach (var camera in cameras)
+                {
+                    Debug.Log(camera);
+                    var cameraDetails = string.Format("Update this camera for HoloLens defaults? \r\nThis will set position to 0,0,0 (on MainCamera only) and background color to Clear Flags:Solid color(Black). \r\n \r\n---------------\r\nName: {0} \r\nPosition: {1}\r\nTag: {2}\r\n---------------",
```

Comment:
In the field, most people I see want project default setups and aren't
using the HoloToolkit yet. Assuming they d/l the toolkit to get this
feature they won't be using the prefab. Also developers I've seen take
existing content and try to bring it to the HoloLens, another case they
wouldn't be using the prefab. They also may have no idea about the prefab.
In that case no reason to replace it, just update the existing camera and
tell them what changed so they are aware of behavior and its not black box.

On Mon, Jul 25, 2016 at 10:00 AM, Neeraj Wadhwa <notifications@github.com>
wrote:

> In Assets/HoloToolkit/Utilities/Editor/ConfigureMenu.cs
> <https://github.com/Microsoft/HoloToolkit-Unity/pull/134#discussion_r72102381>
> :
>
> > -            Camera.main.transform.position = Vector3.zero;
> > -            Camera.main.clearFlags = CameraClearFlags.SolidColor;
> > -            Camera.main.backgroundColor = Color.black;
> > -            Camera.main.nearClipPlane = 0.85f;
> > +            SetCameraDefaults(Camera.main, true, output);
> > +
> > +            //All done with MainCamera
> > +            cameras = cameras.Where(o => o != Camera.main).ToArray();
> > +
> > +            //Check remaining cameras
> > +            if (cameras.Length > 0)
> > +            {
> > +                foreach (var camera in cameras)
> > +                {
> > +                    Debug.Log(camera);
> > +                    var cameraDetails = string.Format("Update this camera for HoloLens defaults? \r\nThis will set position to 0,0,0 (on MainCamera only) and background color to Clear Flags:Solid color(Black). \r\n \r\n---------------\r\nName: {0} \r\nPosition: {1}\r\nTag: {2}\r\n---------------",
>
> I am also not sure if we should be scanning for multiple cameras as only
> those developers who know the specifics really well end up using multiple
> cameras. I would recommend sticking to the main camera prefab from
> HoloToolkit Utilities prefab.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Microsoft/HoloToolkit-Unity/pull/134/files/3933b816638f8cce7b4dc9c87ac4fbcaadee4ded#r72102381>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABqPDXCfL38dR8v-i17IwQyEa6okeBE2ks5qZOu1gaJpZM4JTeQs>
> .
>